src/io/perun.clj

```diff
@@ -177,41 +177,49 @@
        :removed (->> (set/difference x y changed-keys) (select-keys t1) (assoc after :tree))
        :changed (->> changed-keys                      (select-keys t2) (assoc after :tree))})))
 
-(defn- diff
+(defn- fileset-diff
   [before after & props]
   (let [{:keys [added changed]}
         (diff* before after props)]
     (update-in added [:tree] merge (:tree changed))))
 
+;;; end modified boot functions
+
+(defn diff-filesets
+  [before after uses-meta]
+  (if uses-meta
+    ;; Change `fileset-diff` to `boot/fileset-diff` when
+    ;; https://github.com/boot-clj/boot/pull/566 is merged
+    (let [diff (fileset-diff before after :hash pm/+meta-key+)]
+      {:content-diff diff
+       :meta-diff diff})
+    {:content-diff (fileset-diff before after :hash)
+     :meta-diff (fileset-diff before after pm/+meta-key+)}))
```

Comment:
This fixes the main readability issue, but I am still not certain about the names. `content-diff` and `meta-diff` read as the diff of the content and meta respectively, whereas these are files that have changed content and meta respectively.